src/components/Comments/RemoveComment.jsx

```diff
@@ -0,0 +1,48 @@
+import Button from "@mui/material/Button";
+import CircularProgress from "@mui/material/CircularProgress";
+import Box from "@mui/material/Box";
+import { removeComment } from "../api";
+import { useState } from "react";
+
+export default function RemoveComment({
+  commentId,
+  setRenderToggle,
+  setNumberofComments,
+}) {
+  const [loading, setLoading] = useState(false);
+
+  const handleClick = () => {
+    setLoading(true);
+    removeComment(commentId).then(() => {
+      setNumberofComments((currNumber) => {
+        return currNumber - 1;
+      });
+      setRenderToggle((toggle) => {
+        return !toggle;
+      });
+      setLoading(false);
+    });
```

Comment:
This handling of comment deletion works great in terms of making that comment dissapear - but the refresh of the page, then the comment box being opened results in a bit of a glitchy jump when the page reacts to a comment being deleted. Up to you if/how you address this - but wanted to point it out!